crypto/siphash/siphash.c

```diff
@@ -204,7 +204,7 @@ int SipHash_Final(SIPHASH *ctx, unsigned char *out, size_t outlen)
     uint64_t v2 = ctx->v2;
     uint64_t v3 = ctx->v3;
 
-    if (outlen != (size_t)ctx->hash_size)
+    if (ctx->crounds == 0 || outlen == 0 || outlen != (size_t)ctx->hash_size)
```

Comment:
Is failure on `outlen == 0` a common behavior for other `Final` functions, say EVP or other algorithms?

Reply:
I do not think it is a common thing. The SIPHASH has the problem due to the way how it initializes the internal length value.